dapp_runner/runner/runner.py

```diff
@@ -212,13 +255,13 @@ async def _start_service(self, service_name: str, service_descriptor: ServiceDes
                 ]
             )
 
-    async def _start_services(self):
+    async def _start_services(self, resume=False):
         for service_name, service_descriptor in self.dapp.nodes_prioritized():
-            await self._start_service(service_name, service_descriptor)
+            await self._start_service(service_name, service_descriptor, resume=resume)
 
         self._startup_finished = True
 
-    async def start(self):
+    async def start(self, resume=False):
```

Comment:
I would prefer `start`, `_create_networks` and ` _start_service` with `resume=True` to be separated methods. e.g. `resume`, `_resume_networks` and `_resume_service`. 

This would limit the number of times `resume` arg is passed around and frequency of `resume` being checked in `if` statements.

I also wonder if `resume` realy is an options of exciting `start` command or should it be a separated command.

Reply:
as you can see there's still considerable overlap in the logic that happens nevertheless when executing those steps, so in my opinion "resume" still seems like a _mode_ of starting the runner rather than a completely separate operation... 

otherwise, we'll have to pull the logic to a larger number of internal methods, which will make the code more complicated... 

so while I don't right out disagree with you, I feel specifying `resume` as a flag of the binary and a parameter of the various functions that start the components is reasonable and don't see strong arguments to separate them completely...